beacon-chain/forkchoice/doubly-linked-tree/types.go

```diff
@@ -38,29 +40,38 @@ type Store struct {
 	slashedIndices                map[primitives.ValidatorIndex]bool     // the list of equivocating validator indices
 	originRoot                    [fieldparams.RootLength]byte           // The genesis block root
 	genesisTime                   uint64
-	highestReceivedNode           *Node                                      // The highest slot node.
-	receivedBlocksLastEpoch       [fieldparams.SlotsPerEpoch]primitives.Slot // Using `highestReceivedSlot`. The slot of blocks received in the last epoch.
-	allTipsAreInvalid             bool                                       // tracks if all tips are not viable for head
+	highestReceivedNode           *Node                                               // The highest slot node.
+	receivedBlocksLastEpoch       [fieldparams.SlotsPerEpoch]primitives.Slot          // Using `highestReceivedSlot`. The slot of blocks received in the last epoch.
+	allTipsAreInvalid             bool                                                // tracks if all tips are not viable for head
+	PayloadWithholdBoostRoot      [fieldparams.RootLength]byte                        // PayloadWithholdBoostRoot is the root of the block that receives the withhold boost
+	PayloadWithholdBoostFull      bool                                                // PayloadWithholdBoostFull indicates whether the block receiving the withhold boost is full or empty
+	PayloadRevealBoostRoot        [fieldparams.RootLength]byte                        // PayloadRevealBoostRoot is the root of the block that receives the reveal boost
+	ExecutionPayloadStates        map[[fieldparams.RootLength]byte]*state.BeaconState // ExecutionPayloadStates maps block roots to their corresponding beacon states
```

Comment:
We do not cache the beaconstates in forkchoice this would quickly kill the node if there's no finality.

Reply:
removed this